services/121-service/test/visa-card/contact-information-visa-card.test.ts

```diff
@@ -0,0 +1,67 @@
+import { HttpStatus } from '@nestjs/common';
+
+import { SeedScript } from '@121-service/src/scripts/seed-script.enum';
+import {
+  programIdVisa,
+  registrationVisa,
+} from '@121-service/src/seed-data/mock/visa-card.data';
+import { waitFor } from '@121-service/src/utils/waitFor.helper';
+import {
+  seedPaidRegistrations,
+  updateRegistration,
+} from '@121-service/test/helpers/registration.helper';
+import {
+  getAccessToken,
+  resetDB,
+} from '@121-service/test/helpers/utility.helper';
+
+describe('Should update contact information', () => {
+  let accessToken: string;
+
+  beforeEach(async () => {
+    await resetDB(SeedScript.nlrcMultiple);
+    accessToken = await getAccessToken();
+    await waitFor(2_000);
+  });
+
+  it('should update contact information', async () => {
+    // Arrange
+    await seedPaidRegistrations([registrationVisa], programIdVisa);
+
+    // Act
+    const responseUpdateBeforeFspChange = await updateRegistration(
+      programIdVisa,
+      registrationVisa.referenceId,
+      {
+        phoneNumber: '123456789',
+      },
+      'test',
+      accessToken,
+    );
+
+    await updateRegistration(
+      programIdVisa,
+      registrationVisa.referenceId,
+      {
+        programFinancialServiceProviderConfigurationName:
+          'Intersolve-voucher-whatsapp',
+      },
+      'test',
+      accessToken,
+    );
+
+    const responseUpdateAfterFspChange = await updateRegistration(
+      programIdVisa,
+      registrationVisa.referenceId,
+      {
+        phoneNumber: '987654321',
+      },
+      'test',
+      accessToken,
+    );
+
+    // Assert
+    expect(responseUpdateBeforeFspChange.status).toBe(HttpStatus.OK);
+    expect(responseUpdateAfterFspChange.status).toBe(HttpStatus.OK);
```

Comment:
@RubenGeo Can you explain how this setup and assertions test what you want to test? I don't really get it.

Reply:
I updated the labels of describe and it because I found the previous ones inadequate.

This API test was created in response to a bug where we posted:
![image](https://github.com/user-attachments/assets/f8a1d88f-5ef7-4549-a9a1-8975b2d60802)
to Intersolve after a user updated their registration data.

In this PR, I modified our mock service to throw errors when the update contact info endpoints are called with undefined or null. This ensures that the assertions will produce a status code other than 200 if we attempt to call the Intersolve endpoint with undefined values or the string "undefined".

This approach is somewhat suboptimal because it primarily tests whether the PATCH registration endpoint does not return an error, rather than directly testing whether an API call is made to Intersolve. Unfortunately, testing this in an API test is challenging (and, to be fair, we also don’t test this in our do-payment API tests).

I also considered creating a unit test. However, testing this as a “unit” is complicated due to various dependencies, such as whether a registration has an Intersolve customer, what registration data is available, and the FSP configuration associated with the registration. For these reasons, I chose this solution instead. @jannisvisser